### PR TITLE
Spatial flexure reorganisation

### PR DESCRIPTION
**Pull request by rcooke-ast**
This delta PR moves the manual spatial flexure correction to the `RawImage`. The benefit of this approach is that manual spatial flexure can now be assigned for any frame (including calibrations). Previously, it was only science/standard/background frames that could have a manual flexure. Tests pass:

image

### CODE REVIEW

File: pypeit/calibrations.py [hunk 1]
```diff
@@ -381,13 +384,17 @@ def get_arc(self):
...
 
+        # If a manual spatial flexure is requested for any of the frames, collect it now
+        manual_spat_flexure = self.fitstbl.get_shifts(raw_index)
+
         # Otherwise, create the processed file.
         msgs.info(f'Preparing a {frame["class"].calib_type} calibration frame.')
         self.msarc = buildimage.buildimage_fromlist(self.spectrograph, self.det,
                                                     self.par['arcframe'], raw_files,
                                                     bias=self.msbias, bpm=self.msbpm,
                                                     dark=self.msdark, calib_dir=self.calib_dir,
```
Comment (debora-pe) [inline pypeit/calibrations.py]: Found a bug. Obviously not introduced here.

```suggestion
                                                    dark=self.msdark, slits=self.slits, calib_dir=self.calib_dir,
```

File: pypeit/calibrations.py [hunk 2]
```diff
@@ -756,11 +775,12 @@ def get_flats(self):
             = [], None, None, illum_setup, None, detname
```
Comment (debora-pe) [inline pypeit/calibrations.py]: `raw_pixel_index` should be added here too, as an empty list (just for consistency)

File: pypeit/images/buildimage.py [hunk 3]
```diff
@@ -160,7 +160,7 @@ def construct_file_name(cls, calib_key, calib_dir=None, basename=None):
 
 def buildimage_fromlist(spectrograph, det, frame_par, file_list, bias=None, bpm=None, dark=None,
                         scattlight=None, flatimages=None, maxiters=5, ignore_saturation=True,
-                        slits=None, mosaic=None, calib_dir=None, setup=None, calib_id=None):
+                        slits=None, mosaic=None, manual_spat_flexure=None, calib_dir=None, setup=None, calib_id=None):
```
Comment (debora-pe) [inline pypeit/images/buildimage.py:163]: add `manual_spat_flexure` description in the docstring

File: pypeit/images/rawimage.py [hunk 4]
```diff
@@ -401,7 +401,7 @@ def build_rn2img(self, units='e-', digitization=False):
         return np.array(rn2)
 
     def process(self, par, bpm=None, scattlight=None, flatimages=None, bias=None, slits=None, dark=None,
-                mosaic=False, debug=False):
+                mosaic=False, manual_spat_flexure=None, debug=False):
```
Comment (debora-pe) [inline pypeit/images/rawimage.py:404]: add `manual_spat_flexure` description in the docstring

**Review by debora-pe — APPROVED** [review]
@rcooke-ast Thank you for doing this.
I tested the code a bit and it looks good. I only found a bug, which was not introduced here, but your code exposed it! And I have a few more comments/requests.
Please take a look at them, but I am approving anyway.

File: pypeit/calibrations.py (hunk 2 above)
Comment (rcooke-ast) [reply pypeit/calibrations.py]: Agreed - good catch

File: pypeit/images/buildimage.py (hunk 3 above)
Comment (rcooke-ast) [reply pypeit/images/buildimage.py:163]: Thanks, added 👍

File: pypeit/images/rawimage.py (hunk 4 above)
Comment (rcooke-ast) [reply pypeit/images/rawimage.py:404]: Also done  :-)

**Review by rcooke-ast — COMMENTED** [review]
Thanks for the review @debora-pe, and for catching that bug!  All comments addressed, and merging.

**Pull request merged**